lib/chat_api_web/channels/conversation_channel.ex

```diff
@@ -38,6 +39,13 @@ defmodule ChatApiWeb.ConversationChannel do
   # broadcast to everyone in the current topic (conversation:lobby).
   @impl true
   def handle_in("shout", payload, socket) do
+    account = Accounts.get_account!(payload["account_id"])
+
+    account.users
+    |> Enum.filter(fn u -> u.email_alert_on_new_message end)
+    |> Enum.map(fn u -> u.email end)
+    |> Enum.map(fn e -> Email.send(e, payload["body"], payload["conversation_id"]) end)
```

Comment:
can we move all this logic to after line 56 where we create the message in the db? feels like we should create the message before sending the email. also there's no guarantee at this point that the `account_id` or `conversation_id` exist on the payload, so it would be safer to do it within the `case` statement below (though i'm realizing now that maybe a `with` clause would be more idiomatic? not sure...)

Reply:
(also might be nice to create a method to encapsulate `Emails.send_new_message_alerts(conversation_id, message)` or something rather than having all this logic here)

Reply:
Moved it to Emails method to encapsulate it. I think we might have some sort of Alerter that takes a bunch of messaging channels and then picks which ones the user has turned on and sends them.